docs/tools/offline/cli-device-management.md

```diff
@@ -0,0 +1,87 @@
+
+Mbed OS allows firmware update capability, which the Pelion IoT Platform enables. Mbed CLI includes features to prepare and ship updates for devices managed through the [Device Management Portal](https://cloud.mbed.com/docs/current/introduction/index.html). Mbed CLI provides the subcommand `mbed device-management` to manage devices (`mbed dev-mgmt` and `mbed dm` are also available as shorter aliases). The remainder of this document uses the `mbed dm` alias for all device management subcommands. This document explains the steps to enable and use Pelion Device Management with a project.
+
+#### Project setup
+
+Configure your Mbed Cloud SDK API key, target and toolchain. Obtain the API key from the the Device Management Portal.
+
+```
+$ mbed config -G CLOUD_SDK_API_KEY <API_KEY>
+$ mbed target K64F
+$ mbed toolchain GCC_ARM
+```
+
+Initialize the device management feature of Mbed CLI with the following command:
+
+```
+$ mbed dm init -d "<company domain name>" --model-name "<product model identifier>"
+```
+
+<span class="notes">**Note:** If you do not want to enter the subject information for your update certificate (country, state, city, organization and so on), add the `-q` flag to the command above.</span>
+
+This command asks for information about your update certificate. After completing the prompts, Mbed CLI creates several files:
+
+- A certificate in `.update-certificates/default.der`.
+- A matching private key in `.update-certificates/default.key.pem`.
+- A set of default settings in `.manifest_tool.json`.
+- Device Management update credentials in `update_defalut_resources.c`
+- Device Management settings in `.mbed_cloud_config.json`, including default settings for:
+   - A unique vendor identifier, based on the domain name supplied as the `-d` parameter to `mbed dm init`.
+   - A unique model identifier, based on the vendor identifier and the model name supplied as the `--model-name` to `mbed dm init`.
+   - The path of the update certificate and private key.
+- Device Management developer credentials in `mbed_cloud_dev_credentials.c`
+
+<span class="notes">**Note:** The certificate created in `mbed dm init` is not suitable for production. Use it for testing and development only. To create a certificate for production purposes, use an air-gapped computer or a Hardware Security Module. When going to production, conduct a security review on your manifest signing infrastructure because it is the core of the security guarantees for update client.</span>
+
+#### Single-device update
+
+Mbed CLI provides a subcommand, `mbed dm update device`, for development with a device and for testing purposes. After following the steps in [Project setup](#project-setup), perform firmware updates on a device by running:
+
+```
+$ mbed compile
+```
+
+This generates a payload to update the device with. After generating the payload, update the device through Device Management with:
+
+```
+$ mbed dm update device -D <device ID>
+```
```

Comment:
Please update to: `mbed dm update device -D <device ID> -m <target>`
This is a tiny change that makes the life of people a lot easier. Thanks!

Reply:
@MarceloSalazar If you followed the steps, you would not need to add `-m <target>`.

Reply:
Sure, that's an optional configuration of Mbed CLI but not everyone uses it ;-)
Strictly speaking, people need to specify that option when using 'mbed dm update'

Reply:
Then why just `-m`, why not `-t` as well? This explicitly tells you to set target and toolchain, and then uses it.

Reply:
We need `-t` as well and I missed it... needs to be added.
I understand `mbed target` and `mbed toolchain` are recommendations to make things easier for the user, but not requirements.

My intention is to have consistency with docs in [Mbed CLI](https://github.com/ARMmbed/mbed-cli). For example, we have this line *many* times: `mbed compile -t GCC_ARM -m K64F`

Reply:
Very well. Consistency is always a design goal.